packages/amplify-cli-core/src/hooks/hooksExecutor.ts

```diff
@@ -0,0 +1,195 @@
+    return;
+  }
+
+  hooksHandler.setEventPrefix(eventPrefix);
+
+  const projectPath = pathManager.findProjectRoot() ?? process.cwd();
+  const hooksDirPath = pathManager.getHooksDirPath(projectPath);
+  if (!fs.existsSync(hooksDirPath)) {
+    return;
+  }
+
+  const hooksConfig: HooksConfig = stateManager.getHooksConfigJson(projectPath) ?? {};
+
+  const { commandHooksFileMeta, subCommandHooksFileMeta } = getHooksFileMetas(hooksDirPath, hooksHandler.getHooksEvent(), hooksConfig);
+
+  let executionQueue = [commandHooksFileMeta, subCommandHooksFileMeta];
+
+  // merging because we want to remoe as many undeifined values as possible
+  hooksHandler.mergeDataParameter({
+    amplify: {
+      environment: context?.amplify?.getEnvInfo()?.envName ?? undefined,
+      command: hooksHandler.getHooksEvent().command,
+      subCommand: hooksHandler.getHooksEvent().subCommand,
+      argv: hooksHandler.getHooksEvent().argv,
+    },
+  });
+
+  for (let execFileMeta of executionQueue) {
+    if (execFileMeta) {
+      const runtime = getRuntime(execFileMeta, hooksConfig);
+      if (runtime) {
+        await exec(runtime, execFileMeta, hooksHandler.getDataParameter(), errorParameter);
+      }
+    }
+  }
+};
+
+const exec = async (
+  runtime: string,
+  execFileMeta?: HooksFileMeta,
+  dataParameter?: DataParameter,
+  errorParameter?: ErrorParameter,
+): Promise<void> => {
+  if (!execFileMeta?.filePath || !runtime) {
+    return;
+  }
+
+  const projectRoot = pathManager.findProjectRoot() ?? process.cwd();
+  if (!projectRoot) {
+    return;
+  }
+
+  console.log(`\n----- 🪝 ${execFileMeta.baseName} execution start -----`);
+
+  try {
+    logger.info(`hooks file: ${execFileMeta.fileName} execution started`);
+    const childProcess = execa(runtime, [execFileMeta.filePath], {
+      cwd: projectRoot,
+      env: { PATH: process.env.PATH },
+      input: JSON.stringify({
+        data: dataParameter,
+        error: errorParameter,
+      }),
+    });
+    childProcess?.stdout?.pipe(process.stdout);
+    await childProcess;
+    logger.info(`hooks file: ${execFileMeta.fileName} execution ended`);
+  } catch (err) {
+    logger.info(`hooks file: ${execFileMeta.fileName} execution error - ${JSON.stringify(err)}`);
+    if (err?.stderr?.length > 0) {
+      console.error(err.stderr);
+    }
+    if (err?.exitCode) {
+      console.log(`\n${execFileMeta.baseName} hook script exited with exit code ${err.exitCode}`);
+    }
+    console.log('exiting Amplify process...\n');
+    logger.error('hook script exited with error code', err);
+    process.exit(1);
+  }
+
+  console.log(`----- 🪝 ${execFileMeta.baseName} execution end -----\n`);
+};
+
+const getHooksFileMetas = (
+  hooksDirPath: string,
+  hooksEvent: HooksEvent,
+  hooksConfig: HooksConfig,
+): { commandHooksFileMeta?: HooksFileMeta; subCommandHooksFileMeta?: HooksFileMeta } => {
+  const extensionsSupported = getSupportedExtensions(hooksConfig);
+
+  const allFiles = fs
```

Comment:
Wont this return no files if hooksConfig.jsoin is not present?

We also supported a flow where user only put hooks scripts and no hooks file